kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -5953,7 +5953,7 @@ public void commitResource(ContentResourceEdit edit, int priority) throws OverQu
         m_ignoreExtensions = Arrays.asList(ArrayUtils.nullToEmpty(m_serverConfigurationService.getStrings("content.mimeMagic.ignorecontent.extensions")));
         m_ignoreMimeTypes = Arrays.asList(ArrayUtils.nullToEmpty(m_serverConfigurationService.getStrings("content.mimeMagic.ignorecontent.mimetypes")));
 
-        if (m_useMimeMagic && DETECTOR != null && !ResourceProperties.TYPE_URL.equals(currentContentType) && !hasContentTypeAlready) {
+        if (m_useMimeMagic && DETECTOR != null && !ResourceProperties.TYPE_URL.equals(currentContentType) && !hasContentTypeAlready && currentContentType.equals(BasicContentTypeImageService.UNKNOWN_TYPE) ) {
```

Comment:
can you describe this change?

Reply:
Only if the currentContentType is BasicContentTypeImageService.UNKNOWN_TYPE is when we need to call the DETECTOR (even this is not correct at all).

application/x-ipynb+json is changed to text/plain if we don't add this check